src/components/Loader.jsx

```diff
@@ -1,13 +1,18 @@
-import React from 'react';
 import PropTypes from 'prop-types';
 import { ClipLoader, BounceLoader } from 'react-spinners';
 import { Typography } from 'antd';
 import useSWR from 'swr';
-
+import React from 'react';
+import { useSelector } from 'react-redux';
 import fetchAPI from 'utils/http/fetchAPI';
 
 const { Text } = Typography;
-
+const DOWNLOAD_EXPERIMENT = 'download_experiment_s3';
```

Comment:
so I agree that having it repeated is not ideal. However, I did this because of the responsibility principle. How to display the information to the user is the responsibility of the UI, not the worker. If I want to change how information is presented, then I need to change the UI not the worker. Moreover, if you just send the string you can't actually perform different actions (other than displaying the message) which is less flexible.

Reply:
we can't seem to agree. what do you guys think? @cosa65 @Kristian-A

Reply:
I agree with pol.

Worker: gives the ui data
UI: Interprets and displays that data in a way the user understands

If we want to change a text being shown to a user, we would first look in the ui, if we defined it in other services we would need to go hunting across all the repos.

Reply:
I agree that the repeated stuff is annoying, but I think the (eventual) right solution to that should be a shared repo holding the constants that are used in more than one service (and then each of the service repos pulls from that one).

Reply:
The fact that you would have to go to the worker in order to change a message in the UI is enough to convince me that Pol's approach is better.

Reply:
so, while I initially disagreed, after taking a closer look. Currently the errors in the worker work as stefan proposes. We have error classes that define the error code, and the user message like:
```
class WorkResponseError extends Error {
  constructor(errorCode, userMessage, request) {
    super(`${errorCode}: ${userMessage}`);

    this.errorCode = errorCode;
    this.userMessage = userMessage;
    this.request = request;
  }
}
```
the reasoning was that the UI can't know the details of the error, so we already format it in the worker. I think that considering that this is already in place it makes more sense to adapt my PRs to use the same mechanism. Sending a `statusCode` and `userMessage`.

Reply:
ah y, let's leave it as it is then

Reply:
I am still not sure about that. Is it not better to actually adapt this `WorkResponseError` to return an error type (e.g. `error_downloading_files`) and the UI to then decide what to display.

I think the approach you had with the constants is better, and the way it is done in WorkResponseError is suffering from the same issue Martin discussed.

I really don't like the idea to change the worker to update a text field in the UI

Reply:
For me my logic is that the worker knows best what's going on with it. That and the duplicated config files are what makes me prefer my proposed solution.

Reply:
so, the worker sends **both** a status / error code and a friendly user message. I propose to do the same with the status, send a status code and the user message. This would avoid having duplicated code (the status codes) because we can just use the user message (so it's consistent with what's already in the worker) but it's also very prepared to switch into using the codes in the UI whenever we have a centralized repo to avoid duplications.